### PR TITLE
add search comment by article_id and testing

### PR DESCRIPTION
**Pull request by Jord000**
search for comments based on the article_id. provides testing and error handling for incorrect id and for a search without comments

### CODE REVIEW

File: MVC/controllers.js [hunk 1]
```diff
@@ -33,10 +35,26 @@ exports.getArticleById = (req, res, next) => {
...
+  const articleId = req.params.article_id;
+  const promisesInput = [
+    selectArticleById(articleId),
+    selectCommentsById(articleId),
+  ];
+
+  Promise.all(promisesInput)
+    .then((results) => {
+      res.status(200).send({ comments: results[1] })
+    })
```
Comment (johnhutcheon) [inline MVC/controllers.js:55]: Very nicely done! 🏅

File: __tests__/app.test.js [hunk 2]
```diff
@@ -139,3 +139,44 @@ describe('GET:200 /api/articles', () => {
...
+      .then(({ body: { comments } }) => {
+        expect(comments.length).toBe(testFilter.length);
+        comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: expect.any(String),
+            body: expect.any(String),
+            article_id: expect.any(Number)
```
Comment (johnhutcheon) [inline __tests__/app.test.js]: We know the article ID here- so we can expect the specific number 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -139,3 +139,44 @@ describe('GET:200 /api/articles', () => {
...
+      });
+  });
+  test('should handle non-existant ID request', () => {
+    return request(app)
+    .get('/api/articles/99/comments')
+    .expect(404)
+    .then(({ body }) => {
+      expect(body.msg).toEqual('Not Found');
+    });
+  });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:172]: Can we have another error test here, for when we pass an invalid ID, i.e. articles/banana/comments?

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Jordan, nice work! Could you just see my comments and resubmit with the requested test please? Thanks!

**Review by nickkagame — APPROVED** [review]
All good jordan!

**Pull request merged**